responsibleai/responsibleai/rai_insights/rai_base_insights.py

```diff
@@ -0,0 +1,317 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the RAIBaseInsights class."""
+
+import json
+import pickle
+import warnings
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Optional
+
+import pandas as pd
+
+from responsibleai._internal.constants import Metadata
+
+_DATA = 'data'
+_DTYPES = 'dtypes'
+_TRAIN = 'train'
+_TEST = 'test'
+_MODEL = Metadata.MODEL
+_MODEL_PKL = _MODEL + '.pkl'
+_SERIALIZER = 'serializer'
+_MANAGERS = 'managers'
+_JSON_EXTENSION = '.json'
+
+
+class RAIBaseInsights(ABC):
+    """Defines the base class RAIBaseInsights for the top-level API.
+
+    This class is abstract and should not be instantiated.
+    """
+
+    def __init__(self, model: Any, train: pd.DataFrame, test: pd.DataFrame,
+                 target_column: str, task_type: str,
+                 serializer: Optional[Any] = None):
+        """Creates an RAIBaseInsights object.
+
+        :param model: The model to compute RAI insights for.
+            A model that implements sklearn.predict or sklearn.predict_proba
```

Comment:
Have we got all of these nailed down? For example, [LinearRegression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html) doesn't have `predict_proba()` but I'm pretty sure I've used it with `RAIInsights`

Reply:
yes (note in the doc there are multiple possible models specified, including those with only predict functions).
For weird models or models in other frameworks users can use the wrap_model function in the ml-wrappers repository.